velox/dwio/common/MetadataFilter.h

```diff
@@ -46,6 +47,11 @@ class MetadataFilter {
           leafNodeResults,
       std::vector<uint64_t>& finalResult);
 
+  void eval(
+      std::vector<std::pair<const LeafNode*, dwio::common::RowRanges>>&
+          leafNodeResults,
+      dwio::common::RowRanges& finalResult);
```

Comment:
`finalResult` is indices of pages, not rows, using `RowRange` for it is just overkill and duplicate the code without much gain.  Can we just use the existing `std::vector<uint64_t>` version?

Reply:
Then the semantic is different from the above `eval` and we should choose a different name to reflect the difference.